cpp/include/raft/sparse/solver/lanczos.cuh

```diff
@@ -27,6 +28,76 @@ namespace raft::sparse::solver {
 // Eigensolver
 // =========================================================
 
+/**
+ *  @brief Find the smallest eigenpairs using lanczos solver
+ *  @tparam index_type_t the type of data used for indexing.
+ *  @tparam value_type_t the type of data used for weights, distances.
+ *  @param handle the raft handle.
+ *  @param A Sparse matrix in CSR format.
+ *  @param config lanczos config used to set hyperparameters
+ *  @param v0 Initial lanczos vector
+ *  @param eigenvalues output eigenvalues
+ *  @param eigenvectors output eigenvectors
+ *  @return Zero if successful. Otherwise non-zero.
+ */
+template <typename IndexTypeT, typename ValueTypeT>
+auto lanczos_compute_smallest_eigenvectors(
+  raft::resources const& handle,
+  raft::device_csr_matrix_view<ValueTypeT, IndexTypeT, IndexTypeT, IndexTypeT> A,
+  lanczos_solver_config<ValueTypeT> const& config,
+  raft::device_vector_view<ValueTypeT, uint32_t, raft::row_major> v0,
+  raft::device_vector_view<ValueTypeT, uint32_t, raft::col_major> eigenvalues,
+  raft::device_matrix_view<ValueTypeT, uint32_t, raft::col_major> eigenvectors) -> int
+{
+  return detail::lanczos_compute_smallest_eigenvectors<IndexTypeT, ValueTypeT>(
+    handle, A, config, v0, eigenvalues, eigenvectors);
+}
+
+/**
+ *  @brief Find the smallest eigenpairs using lanczos solver
+ *  @tparam index_type_t the type of data used for indexing.
+ *  @tparam value_type_t the type of data used for weights, distances.
+ *  @param handle the raft handle.
+ *  @param rows Vector view of the rows of the sparse matrix.
+ *  @param cols Vector view of the cols of the sparse matrix.
+ *  @param vals Vector view of the vals of the sparse matrix.
+ *  @param config lanczos config used to set hyperparameters
+ *  @param v0 Initial lanczos vector
+ *  @param eigenvalues output eigenvalues
+ *  @param eigenvectors output eigenvectors
+ *  @return Zero if successful. Otherwise non-zero.
+ */
+template <typename IndexTypeT, typename ValueTypeT>
+auto lanczos_compute_smallest_eigenvectors(
+  raft::resources const& handle,
+  raft::device_vector_view<IndexTypeT, uint32_t, raft::row_major> rows,
```

Comment:
We should be exposing public APIs only for our sparse types and not accepting those arrays individually.

Reply:
Looks like this one still needs to be implemented.